pkg/controller/controller.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	framework "github.com/GoogleCloudPlatform/kubernetes/pkg/controller/framework"
+	"time"
+)
+
+// ControllerPlugin is an interface to controller plugins that can be run
+// in kube-controller-manager.  A ControllerPluginManager handles instantiating
+// and managing the lifecycle of controllers.
+type ControllerPlugin interface {
+	// Init initializes the plugin.  This will be called once before Run.
+	Init(host ControllerHost)
+
+	// Name returns the plugin's name.  Plugins should use namespaced names
+	// such as "example.com/replication-controller".  The "kubernetes.io" namespace is
+	// reserved for plugins which are bundled with kubernetes.
+	Name() string
+
+	// DefaultSyncPeriod gets the suggested sync period for the plugin.
+	// The default period is not guaranteed to be honored.
```

Comment:
Can you say more about why it might not be honored, and what bounds the plugin-author can assume?  Example, is this an upper bound on frequency, or a lower bound or neither?  What *are* the bounds?  What if I don't want my plugin running too fast or too slow?